openfe/protocols/openmm_utils/charge_generation.py

```diff
@@ -318,17 +320,21 @@ def assign_offmol_partial_charges(
     Raises
     ------
     ValueError
-      If the ``toolkit_backend`` is not suported by the selected ``method``.
+      If the ``toolkit_backend`` is not supported by the selected ``method``.
       If ``generate_n_conformers`` is ``None``, but the input ``offmol``
       has no associated conformers.
       If the number of conformers passed or generated exceeds the number
       of conformers selected by the partial charge ``method``.
+
+    Returns
+    -------
+     The Molecule with partial charges assigned.
     """
 
     # If you have non-zero charges and not overwriting, just return
     if (offmol.partial_charges is not None and np.any(offmol.partial_charges)):
         if not overwrite:
-            return
+            return offmol
```

Comment:
This is probably going to silently break a bunch of things (particlularly protocols).

Would it be worth doing an `inplace` style flag where we optionally just overwrite the input molecule and otherwise return a molecule?

Reply:
I think that as we still assign the charges to the input molecule and not a copy this should still be fine. I tried to make a small example of this here, which should hopefully mean we don't break anything!
```
from openff.toolkit import Molecule

def charge_mol(mol) -> Molecule:
    mol.assign_partial_charges("am1bcc")
    return mol

m = Molecule.from_smiles("CCO")
charge_mol(m)

print(m.partial_charges)
```
> [-0.13610011111111114 0.12639988888888887 -0.5998001111111111 0.04236688888888887 0.04236688888888887 0.04236688888888887 0.04319988888888887 0.04319988888888887 0.3959998888888889] elementary_charge

Reply:
Yeah it still does in place but I need the return for the multiprocessing to work, we can add an option for none in place but maybe that should be in another PR?